### PR TITLE
Add Jekyll migration tool

### PR DESCRIPTION
**Pull request by PaulAdamDavis**

### CODE REVIEW

**Comment by markstos** [comment]
@PaulAdamDavis Thanks for this. Could you clarify in the docs where the "supplied zip" comes from or the expected format? The `jekyll` command doesn't have a built-in zip export subcommand.

**Comment by markstos** [comment]
From reading the source code, there's a comment that the tool is only known to work on exports from Medium:

https://github.com/TryGhost/migrate/blob/main/packages/mg-jekyll-export/lib/read-zip.js#L11

File: packages/migrate/sources/jekyll.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                try {
+                    ctx.result = mgJSON.toGhostJSON(ctx.result, ctx.options);
+                } catch (error) {
+                    ctx.errors.push(error);
+                    throw error;
+                }
+            }
+        },
+        {
+            title: 'Fetch images via ImageSraper',
```
Comment (markstos) [inline packages/migrate/sources/jekyll.js:134]: s/Sraper/Scraper/

**Comment by markstos** [comment]
It would helpful if the fields recognized from the front matter are documented. 

From looking at the source here, it's clear that `date`, `author` and `title` are recognized.

It's not clear that `tags` from the frontmatter are carried over. Also, I don't see other frontmatter attributes being processed like `permalink`, `basename` and `category`.

File: packages/migrate/sources/jekyll.js (hunk 1 above)
Comment (PaulAdamDavis) [reply packages/migrate/sources/jekyll.js:134]: This was fixed as of https://github.com/TryGhost/migrate/commit/8e50ea12527b8623a9fb6745e8db71431d4db4e8

**Comment by PaulAdamDavis** [comment]
Hi @markstos! I've just pushed the [ability to handle](https://github.com/TryGhost/migrate/commit/b56f05116c4070fe57beb98c0285cfb757e67e64) `tag`, `tags`, `category`, and `categories` front matter.

I've also added a brief explanation of the [expected folder structure](https://github.com/TryGhost/migrate/tree/main/packages/mg-jekyll-export#usage).

This tool was intended to handle a very vanilla Jekyll site, and at the time didn't need to consume `permalink` or `basename` values.

If you can see value in adding support for these, please do [open a new issue](https://github.com/TryGhost/migrate/issues/new) :)
Pull requests are always welcome, and real-world test data we can build & test against is also very welcome!

**Comment by markstos** [comment]
@PaulAdamDavis Thanks for the quick reply and update. I've opened an issue to support `permalink` and `basename` and submitted a PR that will make the `category` the primary tag, which seems like a better choice than the first post tag.

https://github.com/TryGhost/migrate/pull/505

**Pull request merged**